File: front-end/src/app/pages/serviceContract/service-contract.dto.ts
import { Expense } from "../expense/expense";

export enum ContractStatus {
  ACTIVE = 'ACTIVE',
  CLOSED = 'CLOSED',
  CANCELED = 'CANCELED'
}


export interface ServiceContractDTO {
  id: number;
  name: string;
  description?: string;

  supplierId: number;
  supplierName: string;

  categoryId: number;
  categoryName: string;

  totalValue: number;
  paidValue: number;
  remainingValue: number;

  startDate: string; // ISO (yyyy-MM-dd)
  endDate?: string;

  status: ContractStatus;

  payments: Expense[];
}

File: back-end/ConstructionFinance/src/main/java/br/com/galsystem/construction/finance/dto/serviceContract/ServiceContractDTO.java
package br.com.galsystem.construction.finance.dto.serviceContract;

import br.com.galsystem.construction.finance.dto.expense.ExpenseDTO;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

public record ServiceContractDTO(
        Long id,
        Long supplierId,
        String supplierName,
        Long categoryId,
        String categoryName,
        String name,
        String description,
        BigDecimal contractedValue,
        BigDecimal totalPaid,
        BigDecimal balance,
        LocalDate startDate,
        LocalDate endDate,
        List<ExpenseDTO> payments
) {
}

